### DISCUSSION TITLE
Gagal Import Flask (Capstone API)

### DISCUSSION THREAD

**Discussion by vetricko** [teamalgoritma/community#178] [Q&A] [answered]
<img width="431" alt="Screenshot error bike-api" src="https://user-images.githubusercontent.com/65656955/177828415-bb209982-887f-4af1-b257-3631223778de.png">
 
<img width="659" alt="image" src="https://user-images.githubusercontent.com/65656955/177828560-147ec032-0124-4464-b940-4a5e66d0ee8d.png">

boleh tanya, ini pas mengikuti video untuk import Flask lewat vs code keluar error seperti gambar pertama.

<img width="664" alt="image" src="https://user-images.githubusercontent.com/65656955/177828912-32e1a122-05e1-4cf4-89e1-69ea9fb92832.png">

sudah coba python app.py, tertulis No module named 'flask'

**Comment by triarm** [comment]
Hi @vetricko ,
Terkait kendala tersebut, apakah Anda sudah melakukan instalasi package yang dibutuhkan pada environment bike-api? Jika belum silahkan melakukan instalasi package berdasarkan file `requirements.txt` yang sudah disediakan.

Demikian yang dapat kami sampaikan. Semoga membantu.

**Reply by vetricko** [reply]
sudah terinstall semua
<img width="436" alt="image" src="https://user-images.githubusercontent.com/65656955/177933088-051da7a6-e58b-4d83-96f4-5b3c744a0ddb.png">

**Reply by triarm** [reply]
Jika dilihat dari daftar package tersebut, sepertinya package flask belum terinstall pada environment bike-api. Silahkan untuk melakukan instalasi package kembali berdasarkan file `requirements.txt` ya. Semoga menjawab.

**Comment by dwiknrd** [comment, marked as answer]
Hi [vetricko](https://github.com/vetricko) pastikan python interpreter pada vscode sudah pada environment yang baru dibuat ya. 

Berikut adalah tahapan untuk menyetting python interpreter pada vscode:

1. Pada vscode tekan `ctrl+shift+p` untuk menampilkan fitur search
2. Silahkan ketikan "python interpreter"
3. Pilih "Python: Select Interpreter"
4. Pilih interpreter python sesuai dengan environment yang baru dibuat

Semoga membantu :)

**Reply by vetricko** [reply]
baik, barusan saya lakukan kak, masalahnya masih belum diselesaikan
<img width="960" alt="image" src="https://user-images.githubusercontent.com/65656955/177935550-b4ba6a93-64fe-47bb-a098-487fd82426da.png">

**Reply by dwiknrd** [reply]
Baik pak, hal tersebut karena pada environment belum terinstall library flask. Mohon diinstall seperti yang disarankan mas Tria diatas.

**Reply by vetricko** [reply]
<img width="541" alt="image" src="https://user-images.githubusercontent.com/65656955/177940967-9cd2cb27-d5a2-40c9-adcd-c1f2bbe7abac.png">
barusan saya install ulang package requirements dan cek 1 per 1 di notepad. sudah terinstall semua kak

<img width="479" alt="image" src="https://user-images.githubusercontent.com/65656955/177941228-e6e663a1-2773-4ab5-a3f5-598219e8efac.png">
masalah flash nya masih ada di vs code

**Reply by vetricko** [reply]
cara cek sudah terinstall flask atau belum gmn ya

**Reply by dwiknrd** [reply]
Cara check sudah terinstal atau belum anda dapat mengaktifkan environment terlebih dahulu kemudian ketikkan "pip list".
Coba check apakah flask sudah ada pada daftar tersebut atau belum.

**Reply by vetricko** [reply]
baik terimakasih kak, sudah bs jalan setelah import flask secara manual.

informasi tambahan di requirements.txt folder (main-api) tidak ada install library flask

<img width="616" alt="image" src="https://user-images.githubusercontent.com/65656955/177948391-4ce47e10-4a9c-40f1-b5e3-929b01cadca1.png">
